src/Compilers/VisualBasic/Portable/Symbols/MethodSignatureComparer.vb

```diff
@@ -904,6 +904,11 @@ Done:
         ''' </summary>
         Private Shared Function AreConstraintTypesSubset(constraintTypes1 As ArrayBuilder(Of TypeSymbol), constraintTypes2 As ArrayBuilder(Of TypeSymbol)) As Boolean
```

Comment:
>AreConstraintTypesSubset [](start = 32, length = 24)

What happens if a C# 7 or VB 15 compiler encounters the `object` constraint?

Reply:
> What happens if a C# 7 or VB 15 compiler encounters the object constraint in metadata?

C# 7 simply ignores such constraint in metadata (see related changes in PETypeParameterSymbol.cs)
However, VB doesn't do this, which (without the change below) prevented overriding and implementing methods with such constraints because:
- VB disallows explicit Object constraint
- Constraints must be spelled out explicitly and should match overridden/implemented method.

I think the impact on VB should be investigated more, I am planning to open an issue for that. 

---
In reply to: [217126236](https://github.com/dotnet/roslyn/pull/29809#discussion_r217126236) [](ancestors = 217126236)